app/models/concerns/course_user/level_progress_concern.rb

```diff
@@ -0,0 +1,29 @@
+module CourseUser::LevelProgressConcern
+  extend ActiveSupport::Concern
+
+  delegate :level_number, to: :current_level
+
+  # Returns the level object of the CourseUser with respect to a course's Course::Levels.
+  #
+  # @return [Course::Level] Level of CourseUser.
+  def current_level
+    course.level_for(experience_points)
+  end
+
+  # Computes the percentage (a fixnum ranging from 0-100) of the CourseUser's EXP progress
+  # between the current level and the next.  If the CourseUser is at the highest level,
+  # the percentage will be set at 100.
+  #
+  # eg. Current EXP: 500, Level 1 Threshold: 200, Level 2 Threshold: 600
+  # Then CourseUser.level_progress_percentage = 75 # [(500 - 200) / (600 - 200)]
+  #
+  # @return [Fixnum] The CourseUser's EXP progress percentage.
+  def level_progress_percentage
+    if current_level.next
+      100 * (experience_points - current_level.experience_points_threshold) /
+        (current_level.next.experience_points_threshold - current_level.experience_points_threshold)
```

Comment:
Will need to modify the suggested code a little, as CourseUsers at the highest level will get `nil` when trying to access `CourseUser.current_level.next`. 

Intuitively, placing it in both models are all right, although I might prefer it being with the CourseUser. However, since there is this `nil` behaviour arising from the design of levels, would suggest to shift it to the Level Object. 

Will poke you again once I'm done with this @lowjoel